Dart/src/com/jetbrains/lang/dart/psi/impl/DartUriElementBase.java

```diff
@@ -53,6 +57,25 @@ public PsiReference getReference() {
     final String uri = ((DartUriBasedDirective)parent).getUriString();
     final int uriOffset = ((DartUriBasedDirective)parent).getUriStringOffset();
 
+    if (DartResolver.isServerDrivenResolution()) {
+      final PsiFile refPsiFile = getContainingFile();
+      final int refOffset = getTextOffset();
+      final int refLength = getTextLength();
+      final PluginNavigationRegion region = DartResolver.INSTANCE.findRegion(refPsiFile, refOffset, refLength);
+      if (region != null) {
+        final PluginNavigationTarget target = region.getTargets().get(0);
+        final String targetPath = target.getFile();
+        final VirtualFile targetVirtualFile = LocalFileSystem.getInstance().findFileByPath(targetPath);
```

Comment:
Reference should be there even if it can't be resolved. Code that is searching for target file should be moved to DartFileReference.resolve()